lib/api/multipartDelete.js

```diff
@@ -72,11 +72,12 @@ function multipartDelete(authInfo, request, log, callback) {
         function deleteData(mpuBucket, storedParts, next) {
             // The locations were sent to metadata as an array
             // under partLocations.  Pull the partLocations.
-            const locations = storedParts.map(item => item.value.partLocations);
+            let locations = storedParts.map(item => item.value.partLocations);
             if (locations.length === 0) {
                 return next(null, mpuBucket, storedParts);
             }
-            // If have locations, flatten the array
+            // flatten the array
+            locations = [].concat.apply([], locations);
```

Comment:
no.  see the section about apply and built in functions -- http://devdocs.io/javascript/global_objects/function/apply